entities/base_card.rb

```diff
@@ -0,0 +1,54 @@
+class BaseCard
+  attr_accessor :balance
+  attr_reader :type, :number
+
+  def initialize
```

Comment:
def initialize
  @balance = self.class::DEFAULT_BALANCE
end

add to each your card this constant

Reply:
Done